charms/istio-pilot/src/charm.py

```diff
@@ -611,7 +613,7 @@ def _reconcile_ingress_auth(self, ingress_auth_data: dict):
             "app_name": self.app.name,
             "envoyfilter_name": envoyfilter_name,
             "envoyfilter_namespace": self.model.name,
-            "gateway_port": self._gateway_port,
+            "gateway_ports": list(GATEWAY_PORTS.values()),
```

Comment:
What I understand from your PR is that you are passing a list of gateway ports as context to render auth filter manifest file so the things inside the for loop are rendered twice, that way you are applying the EnvoyFilter to both http and https traffic no matter the charm configuration. That being said and just for my own understanding, why do we want to apply the EnvoyFilter to both http and https traffic?

Reply:
> Am I understanding that correctly?

Yes

> Purely out of fear.

If this fix is out of pure fear, then [this](https://discourse.charmhub.io/t/kubeflow-1-7-ssl-and-dex-auth/9574) issue is already fixed in the existing code (before this PR), isn't?

What I am trying to understand is why we need this. In the charm (before this PR), we are supposedly covering both cases, not at the same time, but we know both cases should be covered.

Reply:
After some offline discussions with @DnPlas and @kimwnasptd, we will merge this as is because it is at worst helpful even if it didn't solve the bug, but then I'll dig further to understand exactly what happened separately.

Reply:
After some additional testing, I can confirm

* istio-pilot shipped with Kubeflow 1.6: works as expected
* istio-pilot shipped with Kubeflow 1.7: does not work
* istio-pilot refactored [here](https://github.com/canonical/istio-operators/pull/261) after 1.7: works as expected
* this PR: works as expected

So https://github.com/canonical/bundle-kubeflow/issues/570 was really closed by https://github.com/canonical/istio-operators/pull/261, and this PR adds a bit of safety that prevents a possible problem in future (but one that we've never hit in practice that we know of).

Reply:
Could you please add your findings in the description of this issue (or at least a link to your previous comments) so everyone can see them once this PR is merged?

Reply:
yeah good call.  I'll add a link to this so the whole conversation is easy to spot